src/vivarium/framework/engine.py

```diff
@@ -349,6 +350,26 @@ def get_population(self, untracked: bool = True) -> pd.DataFrame:
     def __repr__(self):
         return f"SimulationContext({self.name})"
 
+    def get_number_of_steps(self, start_time: pd.Timestamp) -> int:
+        time_config = self.configuration.time.to_dict()
+        end_time = pd.Timestamp(**time_config["end"]) if "end" in time_config else None
+        step_size = (
+            pd.Timedelta(days=time_config["step_size"])
+            if "step_size" in time_config
+            else None
+        )
+        if end_time and step_size:
+            num_steps = int(math.ceil((end_time - start_time) / step_size))
+        elif "simulation_events" in time_config:
```

Comment:
Should we handle the event where `simulation_events` AND end time or step size is provided (probably a raise?)?